src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -905,6 +905,12 @@ internal override MemberSemanticModel GetMemberModel(SyntaxNode node)
                                 GetOrAddModelIfContains(propertyDecl.ExpressionBody, span);
                         }
 
+                    case SyntaxKind.DataPropertyDeclaration:
```

Comment:
>SyntaxKind.DataPropertyDeclaration [](start = 25, length = 34)

I think we want to add a strongly typed overload for new node next to
```public static IPropertySymbol? GetDeclaredSymbol(this SemanticModel? semanticModel, PropertyDeclarationSyntax declarationSyntax, CancellationToken cancellationToken = default(CancellationToken))```
and follow the chain if things that that would entail.